sql: unskip TestShowTraceReplica

SHA: https://github.com/cockroachdb/cockroach/commits/295b6ae3142518f04a5771c79ce171043697ee1f

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestShowTraceReplica PKG=github.com/cockroachdb/cockroach/pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1107430&tab=buildLog

```
W190124 06:21:07.618894 129488 server/node.go:869  [n1,summaries] health alerts detected: {Alerts:[{StoreID:1 Category:METRICS Description:queue.replicate.process.failure Value:1 XXX_NoUnkeyedLiteral:{} XXX_sizecache:0}] XXX_NoUnkeyedLiteral:{} XXX_sizecache:0}
I190124 06:21:07.667105 130712 sql/show_trace_replica_test.go:100  SucceedsSoon: UPSERT INTO d.t2 VALUES (1): got [[4 4]] expected [[2 2]]
I190124 06:21:07.726047 129723 server/status/runtime.go:464  [n2] runtime stats: 233 MiB RSS, 1425 goroutines, 71 MiB/38 MiB/140 MiB GO alloc/idle/total, 44 MiB/78 MiB CGO alloc/total, 415.3 CGO/sec, 5.9/1.2 %(u/s)time, 0.0 %gc (1x), 3.7 MiB/3.7 MiB (r/w)net
I190124 06:21:07.814625 129969 server/status/runtime.go:464  [n3] runtime stats: 233 MiB RSS, 1425 goroutines, 41 MiB/62 MiB/140 MiB GO alloc/idle/total, 44 MiB/78 MiB CGO alloc/total, 411.7 CGO/sec, 5.7/1.2 %(u/s)time, 0.0 %gc (1x), 3.7 MiB/3.7 MiB (r/w)net
W190124 06:21:07.822366 129987 server/node.go:869  [n3,summaries] health alerts detected: {Alerts:[{StoreID:3 Category:METRICS Description:queue.replicate.process.failure Value:3 XXX_NoUnkeyedLiteral:{} XXX_sizecache:0}] XXX_NoUnkeyedLiteral:{} XXX_sizecache:0}
I190124 06:21:07.889495 130252 server/status/runtime.go:464  [n4] runtime stats: 235 MiB RSS, 1425 goroutines, 48 MiB/55 MiB/140 MiB GO alloc/idle/total, 46 MiB/81 MiB CGO alloc/total, 411.4 CGO/sec, 5.7/1.1 %(u/s)time, 0.0 %gc (1x), 3.7 MiB/3.7 MiB (r/w)net
W190124 06:21:07.893851 130254 server/node.go:869  [n4,summaries] health alerts detected: {Alerts:[{StoreID:4 Category:METRICS Description:queue.replicate.process.failure Value:2 XXX_NoUnkeyedLiteral:{} XXX_sizecache:0}] XXX_NoUnkeyedLiteral:{} XXX_sizecache:0}
I190124 06:21:08.671340 130712 sql/show_trace_replica_test.go:100  SucceedsSoon: UPSERT INTO d.t2 VALUES (1): got [[4 4]] expected [[2 2]]
I190124 06:21:09.675203 130712 sql/show_trace_replica_test.go:100  SucceedsSoon: UPSERT INTO d.t2 VALUES (1): got [[4 4]] expected [[2 2]]
I190124 06:21:10.679701 130712 sql/show_trace_replica_test.go:100  SucceedsSoon: UPSERT INTO d.t2 VALUES (1): got [[4 4]] expected [[2 2]]
I190124 06:21:11.683327 130712 sql/show_trace_replica_test.go:100  SucceedsSoon: UPSERT INTO d.t2 VALUES (1): got [[4 4]] expected [[2 2]]
I190124 06:21:11.908191 129459 storage/compactor/compactor.go:374  [n1,s1,compactor] processing compaction #1-8/8 (/Table/SystemConfigSpan/Start-/Max) for 43 KiB (reasons: size=false used=true avail=false)
I190124 06:21:11.911505 129459 storage/compactor/compactor.go:390  [n1,s1,compactor] processed compaction #1-8/8 (/Table/SystemConfigSpan/Start-/Max) for 43 KiB in 0.0s
I190124 06:21:12.579277 130246 storage/compactor/compactor.go:329  [n4,s4,compactor] purging suggested compaction for range /Min - /System/NodeLiveness that contains live data
I190124 06:21:12.687854 130712 sql/show_trace_replica_test.go:100  SucceedsSoon: UPSERT INTO d.t2 VALUES (1): got [[4 4]] expected [[2 2]]
    --- FAIL: TestShowTraceReplica/UPSERT_INTO_d.t2_VALUES_(1) (45.43s)
        show_trace_replica_test.go:100: condition failed to evaluate within 45s: UPSERT INTO d.t2 VALUES (1): got [[4 4]] expected [[2 2]]
            goroutine 130712 [running]:
            runtime/debug.Stack(0xa7a358200, 0xc001876120, 0x3298e80)
            	/usr/local/go/src/runtime/debug/stack.go:24 +0xa7
            github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x32fbee0, 0xc0050f0300, 0xc003d7fca0)
            	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x103
            github.com/cockroachdb/cockroach/pkg/sql_test.TestShowTraceReplica.func2(0xc0050f0300)
            	/go/src/github.com/cockroachdb/cockroach/pkg/sql/show_trace_replica_test.go:100 +0x90
            testing.tRunner(0xc0050f0300, 0xc003d7fc60)
            	/usr/local/go/src/testing/testing.go:827 +0xbf
            created by testing.(*T).Run
            	/usr/local/go/src/testing/testing.go:878 +0x35c

```

Jira issue: CRDB-4661

Epic CRDB-27234

SHA: https://github.com/cockroachdb/cockroach/commits/295b6ae3142518f04a5771c79ce171043697ee1f

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestShowTraceReplica PKG=github.com/cockroachdb/cockroach/pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1107708&tab=buildLog

```
I190124 08:45:18.914981 128383 storage/store_snapshot.go:805  [n4,replicate,s4,r22/2:/Table/5{4-5}] streamed snapshot to (n2,s2):?: kv pairs: 77, log entries: 89, rate-limit: 2.0 MiB/sec, 0.01s
I190124 08:45:18.915460 128384 storage/replica_destroy.go:152  [n1,replicaGC,s1,r22/1:/Table/5{4-5}] removed 8 (1+7) keys in 1ms [clear=1ms commit=0ms]
I190124 08:45:18.922765 128422 storage/replica_raftstorage.go:805  [n2,s2,r22/?:{-}] applying preemptive snapshot at index 99 (id=37616f5f, encoded size=12494, 1 rocksdb batches, 89 log entries)
I190124 08:45:18.943595 128422 storage/replica_raftstorage.go:811  [n2,s2,r22/?:/Table/5{4-5}] applied preemptive snapshot in 20ms [clear=0ms batch=0ms entries=18ms commit=1ms]
I190124 08:45:18.956631 128383 storage/replica_command.go:798  [n4,replicate,s4,r22/2:/Table/5{4-5}] change replicas (ADD_REPLICA (n2,s2):3): read existing descriptor r22:/Table/5{4-5} [(n4,s4):2, next=3, gen=1]
I190124 08:45:19.007278 128383 storage/replica_raft.go:372  [n4,s4,r22/2:/Table/5{4-5}] proposing ADD_REPLICA((n2,s2):3): updated=[(n4,s4):2 (n2,s2):3] next=4
I190124 08:45:19.743759 124557 server/status/runtime.go:464  [n1] runtime stats: 1.2 GiB RSS, 1420 goroutines, 68 MiB/41 MiB/140 MiB GO alloc/idle/total, 142 MiB/192 MiB CGO alloc/total, 396.0 CGO/sec, 98.4/5.2 %(u/s)time, 0.3 %gc (4x), 2.7 MiB/2.7 MiB (r/w)net
I190124 08:45:19.893467 126335 sql/show_trace_replica_test.go:100  SucceedsSoon: UPSERT INTO d.t2 VALUES (1): got [[4 4]] expected [[2 2]]
I190124 08:45:20.871283 128440 storage/replica_command.go:798  [n4,replicate,s4,r19/3:/Table/2{3-4}] change replicas (REMOVE_REPLICA (n1,s1):1): read existing descriptor r19:/Table/2{3-4} [(n1,s1):1, (n4,s4):3, next=4, gen=0]
I190124 08:45:20.920634 128440 storage/replica_raft.go:372  [n4,s4,r19/3:/Table/2{3-4}] proposing REMOVE_REPLICA((n1,s1):1): updated=[(n4,s4):3] next=4
I190124 08:45:20.956266 126335 sql/show_trace_replica_test.go:100  SucceedsSoon: UPSERT INTO d.t2 VALUES (1): got [[4 4]] expected [[2 2]]
I190124 08:45:21.016275 128424 storage/store.go:2656  [n1,replicaGC,s1,r19/1:/Table/2{3-4}] removing replica r19/1
I190124 08:45:21.018841 128424 storage/replica_destroy.go:152  [n1,replicaGC,s1,r19/1:/Table/2{3-4}] removed 11 (4+7) keys in 1ms [clear=1ms commit=0ms]
I190124 08:45:21.289654 124860 server/status/runtime.go:464  [n2] runtime stats: 1.2 GiB RSS, 1420 goroutines, 57 MiB/52 MiB/140 MiB GO alloc/idle/total, 142 MiB/192 MiB CGO alloc/total, 302.0 CGO/sec, 96.6/4.9 %(u/s)time, 0.1 %gc (4x), 2.4 MiB/2.4 MiB (r/w)net
I190124 08:45:21.897899 125178 server/status/runtime.go:464  [n3] runtime stats: 1.2 GiB RSS, 1420 goroutines, 61 MiB/49 MiB/140 MiB GO alloc/idle/total, 142 MiB/192 MiB CGO alloc/total, 306.2 CGO/sec, 94.3/4.7 %(u/s)time, 0.1 %gc (4x), 2.3 MiB/2.3 MiB (r/w)net
I190124 08:45:22.004341 126335 sql/show_trace_replica_test.go:100  SucceedsSoon: UPSERT INTO d.t2 VALUES (1): got [[4 4]] expected [[2 2]]
I190124 08:45:22.681923 125628 server/status/runtime.go:464  [n4] runtime stats: 1.2 GiB RSS, 1420 goroutines, 44 MiB/68 MiB/140 MiB GO alloc/idle/total, 142 MiB/192 MiB CGO alloc/total, 264.5 CGO/sec, 90.5/4.3 %(u/s)time, 0.0 %gc (4x), 2.1 MiB/2.1 MiB (r/w)net
    --- FAIL: TestShowTraceReplica/UPSERT_INTO_d.t2_VALUES_(1) (45.03s)
        <autogenerated>:1: condition failed to evaluate within 45s: UPSERT INTO d.t2 VALUES (1): got [[4 4]] expected [[2 2]]
            goroutine 126335 [running]:
            runtime/debug.Stack(0xa7a358200, 0xc0017ab110, 0x47ae2a0)
            	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
            github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x481b3a0, 0xc006786400, 0xc004bce300)
            	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
            github.com/cockroachdb/cockroach/pkg/sql_test.TestShowTraceReplica.func2(0xc006786400)
            	/go/src/github.com/cockroachdb/cockroach/pkg/sql/show_trace_replica_test.go:100 +0x11b
            testing.tRunner(0xc006786400, 0xc004bce2c0)
            	/usr/local/go/src/testing/testing.go:827 +0x163
            created by testing.(*T).Run
            	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/dc2fbcdc0dccb8cc676fc67370375bab36b3cff0

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestShowTraceReplica PKG=github.com/cockroachdb/cockroach/pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1110826&tab=buildLog

```
I190126 08:43:52.367628 127681 storage/replica_raftstorage.go:805  [n4,s4,r13/?:{-}] applying preemptive snapshot at index 28 (id=2f0f57bd, encoded size=3238, 1 rocksdb batches, 18 log entries)
I190126 08:43:52.372350 127681 storage/replica_raftstorage.go:811  [n4,s4,r13/?:/Table/1{7-8}] applied preemptive snapshot in 4ms [clear=0ms batch=0ms entries=2ms commit=0ms]
I190126 08:43:52.378981 127634 storage/replica_command.go:798  [n1,replicate,s1,r13/1:/Table/1{7-8}] change replicas (ADD_REPLICA (n4,s4):3): read existing descriptor r13:/Table/1{7-8} [(n1,s1):1, next=3, gen=0]
I190126 08:43:52.576190 127634 storage/replica_raft.go:372  [n1,s1,r13/1:/Table/1{7-8}] proposing ADD_REPLICA((n4,s4):3): updated=[(n1,s1):1 (n4,s4):3] next=4
I190126 08:43:53.196698 125914 sql/show_trace_replica_test.go:100  SucceedsSoon: UPSERT INTO d.t2 VALUES (1): got [[1 1]] expected [[2 2]]
I190126 08:43:53.919401 123753 server/status/runtime.go:464  [n1] runtime stats: 1.1 GiB RSS, 1422 goroutines, 43 MiB/64 MiB/140 MiB GO alloc/idle/total, 143 MiB/198 MiB CGO alloc/total, 280.9 CGO/sec, 94.0/5.0 %(u/s)time, 0.0 %gc (5x), 2.4 MiB/2.4 MiB (r/w)net
I190126 08:43:54.231370 125914 sql/show_trace_replica_test.go:100  SucceedsSoon: UPSERT INTO d.t2 VALUES (1): got [[1 1]] expected [[2 2]]
W190126 08:43:54.264970 123755 server/node.go:869  [n1,summaries] health alerts detected: {Alerts:[{StoreID:1 Category:METRICS Description:queue.replicate.process.failure Value:3 XXX_NoUnkeyedLiteral:{} XXX_sizecache:0}] XXX_NoUnkeyedLiteral:{} XXX_sizecache:0}
I190126 08:43:55.198410 124293 server/status/runtime.go:464  [n2] runtime stats: 1.1 GiB RSS, 1422 goroutines, 64 MiB/46 MiB/140 MiB GO alloc/idle/total, 143 MiB/198 MiB CGO alloc/total, 215.5 CGO/sec, 90.5/4.5 %(u/s)time, 0.0 %gc (4x), 2.5 MiB/2.5 MiB (r/w)net
I190126 08:43:55.284822 125914 sql/show_trace_replica_test.go:100  SucceedsSoon: UPSERT INTO d.t2 VALUES (1): got [[1 1]] expected [[2 2]]
I190126 08:43:55.756417 124417 server/status/runtime.go:464  [n3] runtime stats: 1.1 GiB RSS, 1422 goroutines, 40 MiB/68 MiB/140 MiB GO alloc/idle/total, 143 MiB/198 MiB CGO alloc/total, 187.7 CGO/sec, 94.3/4.6 %(u/s)time, 0.0 %gc (5x), 2.5 MiB/2.5 MiB (r/w)net
I190126 08:43:56.322055 125914 sql/show_trace_replica_test.go:100  SucceedsSoon: UPSERT INTO d.t2 VALUES (1): got [[1 1]] expected [[2 2]]
I190126 08:43:56.396960 124884 server/status/runtime.go:464  [n4] runtime stats: 1.1 GiB RSS, 1422 goroutines, 54 MiB/55 MiB/140 MiB GO alloc/idle/total, 143 MiB/198 MiB CGO alloc/total, 190.4 CGO/sec, 89.2/4.7 %(u/s)time, 0.0 %gc (4x), 2.6 MiB/2.6 MiB (r/w)net
W190126 08:43:56.949790 124886 server/node.go:869  [n4,summaries] health alerts detected: {Alerts:[{StoreID:4 Category:METRICS Description:queue.replicate.process.failure Value:3 XXX_NoUnkeyedLiteral:{} XXX_sizecache:0}] XXX_NoUnkeyedLiteral:{} XXX_sizecache:0}
I190126 08:43:57.368060 125914 sql/show_trace_replica_test.go:100  SucceedsSoon: UPSERT INTO d.t2 VALUES (1): got [[1 1]] expected [[2 2]]
I190126 08:43:58.415698 125914 sql/show_trace_replica_test.go:100  SucceedsSoon: UPSERT INTO d.t2 VALUES (1): got [[1 1] [4 4]] expected [[2 2]]
    --- FAIL: TestShowTraceReplica/UPSERT_INTO_d.t2_VALUES_(1) (45.89s)
        <autogenerated>:1: condition failed to evaluate within 45s: UPSERT INTO d.t2 VALUES (1): got [[1 1] [4 4]] expected [[2 2]]
            goroutine 125914 [running]:
            runtime/debug.Stack(0xa7a358200, 0xc000ea7ef0, 0x47acb40)
            	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
            github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4819e20, 0xc001376300, 0xc001de3ee0)
            	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
            github.com/cockroachdb/cockroach/pkg/sql_test.TestShowTraceReplica.func2(0xc001376300)
            	/go/src/github.com/cockroachdb/cockroach/pkg/sql/show_trace_replica_test.go:100 +0x11b
            testing.tRunner(0xc001376300, 0xc001de3e80)
            	/usr/local/go/src/testing/testing.go:827 +0x163
            created by testing.(*T).Run
            	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/9f084ad576e85756c5c5a7e41335d9aa2d3eee30

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestShowTraceReplica PKG=github.com/cockroachdb/cockroach/pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1112213&tab=buildLog

```
I190128 06:17:51.923754 132885 storage/replica_command.go:798  [n4,replicate,s4,r4/3:/System/ts{d-e}] change replicas (REMOVE_REPLICA (n1,s1):1): read existing descriptor r4:/System/ts{d-e} [(n1,s1):1, (n4,s4):3, next=4, gen=0]
I190128 06:17:51.927465 132885 storage/replica_raft.go:372  [n4,s4,r4/3:/System/ts{d-e}] proposing REMOVE_REPLICA((n1,s1):1): updated=[(n4,s4):3] next=4
I190128 06:17:51.929190 132844 storage/store.go:2669  [n1,replicaGC,s1,r4/1:/System/ts{d-e}] removing replica r4/1
I190128 06:17:51.929913 132844 storage/replica_destroy.go:152  [n1,replicaGC,s1,r4/1:/System/ts{d-e}] removed 2080 (2072+8) keys in 1ms [clear=1ms commit=0ms]
I190128 06:17:52.798355 130926 sql/show_trace_replica_test.go:100  SucceedsSoon: UPSERT INTO d.t2 VALUES (1): got [[4 4]] expected [[2 2]]
I190128 06:17:53.801756 130926 sql/show_trace_replica_test.go:100  SucceedsSoon: UPSERT INTO d.t2 VALUES (1): got [[4 4]] expected [[2 2]]
I190128 06:17:53.924980 132870 storage/replica_command.go:798  [n4,replicate,s4,r12/3:/Table/1{6-7}] change replicas (REMOVE_REPLICA (n1,s1):1): read existing descriptor r12:/Table/1{6-7} [(n1,s1):1, (n4,s4):3, next=4, gen=0]
I190128 06:17:53.928278 132870 storage/replica_raft.go:372  [n4,s4,r12/3:/Table/1{6-7}] proposing REMOVE_REPLICA((n1,s1):1): updated=[(n4,s4):3] next=4
I190128 06:17:53.929812 132950 storage/store.go:2669  [n1,replicaGC,s1,r12/1:/Table/1{6-7}] removing replica r12/1
I190128 06:17:53.930842 132950 storage/replica_destroy.go:152  [n1,replicaGC,s1,r12/1:/Table/1{6-7}] removed 7 (0+7) keys in 1ms [clear=0ms commit=1ms]
I190128 06:17:54.805267 130926 sql/show_trace_replica_test.go:100  SucceedsSoon: UPSERT INTO d.t2 VALUES (1): got [[4 4]] expected [[2 2]]
I190128 06:17:54.981605 132954 storage/replica_command.go:798  [n4,replicate,s4,r19/3:/Table/2{3-4}] change replicas (REMOVE_REPLICA (n2,s2):2): read existing descriptor r19:/Table/2{3-4} [(n2,s2):2, (n4,s4):3, next=4, gen=0]
I190128 06:17:54.984677 132954 storage/replica_raft.go:372  [n4,s4,r19/3:/Table/2{3-4}] proposing REMOVE_REPLICA((n2,s2):2): updated=[(n4,s4):3] next=4
I190128 06:17:55.040799 132903 storage/store.go:2669  [n2,replicaGC,s2,r19/2:/Table/2{3-4}] removing replica r19/2
I190128 06:17:55.041050 132903 storage/replica_destroy.go:152  [n2,replicaGC,s2,r19/2:/Table/2{3-4}] removed 11 (4+7) keys in 0ms [clear=0ms commit=0ms]
I190128 06:17:55.808867 130926 sql/show_trace_replica_test.go:100  SucceedsSoon: UPSERT INTO d.t2 VALUES (1): got [[4 4]] expected [[2 2]]
I190128 06:17:55.925491 132965 storage/replica_command.go:798  [n4,replicate,s4,r8/3:/Table/1{2-3}] change replicas (REMOVE_REPLICA (n1,s1):1): read existing descriptor r8:/Table/1{2-3} [(n1,s1):1, (n4,s4):3, next=4, gen=0]
I190128 06:17:55.928548 132965 storage/replica_raft.go:372  [n4,s4,r8/3:/Table/1{2-3}] proposing REMOVE_REPLICA((n1,s1):1): updated=[(n4,s4):3] next=4
I190128 06:17:55.930199 132911 storage/store.go:2669  [n1,replicaGC,s1,r8/1:/Table/1{2-3}] removing replica r8/1
I190128 06:17:55.930614 132911 storage/replica_destroy.go:152  [n1,replicaGC,s1,r8/1:/Table/1{2-3}] removed 102 (95+7) keys in 0ms [clear=0ms commit=0ms]
    --- FAIL: TestShowTraceReplica/UPSERT_INTO_d.t2_VALUES_(1) (45.40s)
        show_trace_replica_test.go:100: condition failed to evaluate within 45s: UPSERT INTO d.t2 VALUES (1): got [[4 4]] expected [[2 2]]
            goroutine 130926 [running]:
            runtime/debug.Stack(0xa7a358200, 0xc003cca8d0, 0x327a540)
            	/usr/local/go/src/runtime/debug/stack.go:24 +0xa7
            github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x32dd7e0, 0xc002dab700, 0xc001b6c1c0)
            	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x103
            github.com/cockroachdb/cockroach/pkg/sql_test.TestShowTraceReplica.func2(0xc002dab700)
            	/go/src/github.com/cockroachdb/cockroach/pkg/sql/show_trace_replica_test.go:100 +0x90
            testing.tRunner(0xc002dab700, 0xc001b6c180)
            	/usr/local/go/src/testing/testing.go:827 +0xbf
            created by testing.(*T).Run
            	/usr/local/go/src/testing/testing.go:878 +0x35c

```

SHA: https://github.com/cockroachdb/cockroach/commits/9f084ad576e85756c5c5a7e41335d9aa2d3eee30

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestShowTraceReplica PKG=github.com/cockroachdb/cockroach/pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1112491&tab=buildLog

```
I190128 09:38:04.215219 126534 sql/show_trace_replica_test.go:100  SucceedsSoon: UPSERT INTO d.t2 VALUES (1): got [[4 4]] expected [[2 2]]
I190128 09:38:05.256403 126534 sql/show_trace_replica_test.go:100  SucceedsSoon: UPSERT INTO d.t2 VALUES (1): got [[4 4]] expected [[2 2]]
I190128 09:38:05.611817 128324 storage/replica_command.go:798  [n3,replicate,s3,r23/3:/{Table/55-Max}] change replicas (REMOVE_REPLICA (n4,s4):2): read existing descriptor r23:/{Table/55-Max} [(n4,s4):2, (n3,s3):3, next=4, gen=0]
I190128 09:38:05.731428 128324 storage/replica_raft.go:372  [n3,s3,r23/3:/{Table/55-Max}] proposing REMOVE_REPLICA((n4,s4):2): updated=[(n3,s3):3] next=4
I190128 09:38:05.755338 128273 storage/store.go:2669  [n4,replicaGC,s4,r23/2:/{Table/55-Max}] removing replica r23/2
I190128 09:38:05.761947 128273 storage/replica_destroy.go:152  [n4,replicaGC,s4,r23/2:/{Table/55-Max}] removed 7 (0+7) keys in 1ms [clear=0ms commit=0ms]
I190128 09:38:06.321108 126534 sql/show_trace_replica_test.go:100  SucceedsSoon: UPSERT INTO d.t2 VALUES (1): got [[4 4]] expected [[2 2]]
I190128 09:38:07.364699 126534 sql/show_trace_replica_test.go:100  SucceedsSoon: UPSERT INTO d.t2 VALUES (1): got [[4 4]] expected [[2 2]]
I190128 09:38:07.715333 124524 server/status/runtime.go:464  [n1] runtime stats: 1.1 GiB RSS, 1412 goroutines, 64 MiB/45 MiB/140 MiB GO alloc/idle/total, 144 MiB/193 MiB CGO alloc/total, 563.8 CGO/sec, 102.8/6.1 %(u/s)time, 0.3 %gc (4x), 1.4 MiB/1.4 MiB (r/w)net
I190128 09:38:08.415813 126534 sql/show_trace_replica_test.go:100  SucceedsSoon: UPSERT INTO d.t2 VALUES (1): got [[4 4]] expected [[2 2]]
I190128 09:38:09.063966 125051 server/status/runtime.go:464  [n2] runtime stats: 1.2 GiB RSS, 1412 goroutines, 50 MiB/58 MiB/140 MiB GO alloc/idle/total, 146 MiB/193 MiB CGO alloc/total, 283.6 CGO/sec, 95.3/5.5 %(u/s)time, 0.3 %gc (4x), 1.3 MiB/1.3 MiB (r/w)net
I190128 09:38:09.461881 126534 sql/show_trace_replica_test.go:100  SucceedsSoon: UPSERT INTO d.t2 VALUES (1): got [[4 4]] expected [[2 2]]
I190128 09:38:09.528921 125332 server/status/runtime.go:464  [n3] runtime stats: 1.2 GiB RSS, 1412 goroutines, 65 MiB/45 MiB/140 MiB GO alloc/idle/total, 146 MiB/193 MiB CGO alloc/total, 284.9 CGO/sec, 95.9/5.2 %(u/s)time, 0.3 %gc (4x), 1.2 MiB/1.2 MiB (r/w)net
W190128 09:38:09.806203 125334 server/node.go:869  [n3,summaries] health alerts detected: {Alerts:[{StoreID:3 Category:METRICS Description:queue.replicate.process.failure Value:2 XXX_NoUnkeyedLiteral:{} XXX_sizecache:0}] XXX_NoUnkeyedLiteral:{} XXX_sizecache:0}
I190128 09:38:10.116155 125658 server/status/runtime.go:464  [n4] runtime stats: 1.2 GiB RSS, 1412 goroutines, 73 MiB/37 MiB/140 MiB GO alloc/idle/total, 148 MiB/195 MiB CGO alloc/total, 232.9 CGO/sec, 91.9/4.7 %(u/s)time, 0.2 %gc (3x), 1.1 MiB/1.1 MiB (r/w)net
I190128 09:38:10.522224 126534 sql/show_trace_replica_test.go:100  SucceedsSoon: UPSERT INTO d.t2 VALUES (1): got [[4 4]] expected [[2 2]]
W190128 09:38:10.773970 125660 server/node.go:869  [n4,summaries] health alerts detected: {Alerts:[{StoreID:4 Category:METRICS Description:queue.replicate.process.failure Value:1 XXX_NoUnkeyedLiteral:{} XXX_sizecache:0}] XXX_NoUnkeyedLiteral:{} XXX_sizecache:0}
    --- FAIL: TestShowTraceReplica/UPSERT_INTO_d.t2_VALUES_(1) (45.86s)
        <autogenerated>:1: condition failed to evaluate within 45s: UPSERT INTO d.t2 VALUES (1): got [[4 4]] expected [[2 2]]
            goroutine 126534 [running]:
            runtime/debug.Stack(0xa7a358200, 0xc006fc2510, 0x47a9520)
            	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
            github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4816860, 0xc0052c6800, 0xc002addae0)
            	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
            github.com/cockroachdb/cockroach/pkg/sql_test.TestShowTraceReplica.func2(0xc0052c6800)
            	/go/src/github.com/cockroachdb/cockroach/pkg/sql/show_trace_replica_test.go:100 +0x11b
            testing.tRunner(0xc0052c6800, 0xc002addaa0)
            	/usr/local/go/src/testing/testing.go:827 +0x163
            created by testing.(*T).Run
            	/usr/local/go/src/testing/testing.go:878 +0x65a

```

--- a/pkg/sql/show_trace_replica_test.go
+++ b/pkg/sql/show_trace_replica_test.go
@@ -21,7 +21,6 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
 	"github.com/cockroachdb/cockroach/pkg/server"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
-	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/testcluster"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
@@ -33,8 +32,6 @@ import (
 func TestShowTraceReplica(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-
-	skip.WithIssue(t, 34213)
 
 	const numNodes = 4
 

